src/frameworks/next/index.ts

```diff
@@ -177,18 +187,18 @@ export async function build(dir: string): Promise<BuildResult> {
       // Can we change i18n into Firebase settings?
       if (has) return false;
 
-      if (
-        supportsFrameworkRegex(source) &&
-        supportsFrameworkRegex(destination) &&
-        isUrl(destination) === false
-      ) {
-        return true;
-      } else {
+      if (pathHasRegex(source) || isUrl(destination)) {
```

Comment:
Maybe we could break these into util functions? `filter(rewriteSupportedByFirebase)` that would make them easier to test & reuse in codegenPublicDir too.

would you mind taking on the work to utilize in codegenPublicDir in this PR? the idea being if the regex matches the path (for !supportedByFirebase) we should not copy that file into the CDN